packages/teleport/src/services/recordings/makeRecording.ts

```diff
@@ -1,35 +1,43 @@
 import moment from 'moment';
-import { Recording } from './types';
+import { Recording, RecordingType } from './types';
+import { eventCodes } from 'teleport/services/audit';
 
+// Takes in json objects built by SessionEnd and WindowsDesktopSessionEnd as defined in teleport/api/types/events/events.proto.
 export default function makeRecording({
   participants = [],
   time,
   session_start,
   session_stop,
   server_hostname,
   interactive,
+  user,
   session_recording = 'on',
   sid,
   proto = '',
   kubernetes_cluster = '',
   kubernetes_pod_namespace = '',
   kubernetes_pod_name = '',
+  desktop_name = '',
+  code,
 }): Recording {
+  let type: RecordingType =
+    code === eventCodes.DESKTOP_SESSION_ENDED ? 'desktop' : 'ssh';
   let durationText = '';
   let duration = 0;
   if (session_start && session_stop) {
     duration = moment(session_stop).diff(session_start);
     durationText = moment.duration(duration).humanize();
   }
 
-  let hostname = server_hostname || 'N/A';
+  // Desktop sessions will have desktop_name set, ssh sessions can have server_hostname or neither.
```

Comment:
I inferred "ssh sessions can have server_hostname or neither" from the logic, though I'm not sure in what situation an ssh session wouldn't have a server_hostname.

Reply:
I think so, though I'm not sure precisely where the names are coming from. cc @zmb3

Reply:
@zmb3 consider using similar property names for desktop events. For example `user` could be a `participant` - these things will make it a bit easier for people to consume the events.